modules/gapi/src/backends/onnx/gonnxbackend.cpp

```diff
@@ -451,8 +490,15 @@ void cv::gimpl::onnx::GONNXExecutable::run(std::vector<InObj>  &&input_objs,
     context.args.reserve(op.args.size());
     using namespace std::placeholders;
     ade::util::transform(op.args,
-                          std::back_inserter(context.args),
-                          std::bind(&GONNXExecutable::packArg, this, _1));
+                         std::back_inserter(context.args),
+                         std::bind(&GONNXExecutable::packArg, this, _1));
+
+    // NB: Need to store inputs shape to recognize GFrame/GMat
+    ade::util::transform(op.args,
+                         std::back_inserter(context.in_shapes),
```

Comment:
Ideally `in_shapes` must be `reserve()`-d like `context.args` above.

Reply:
Fixed.